### PR TITLE
#167727330 Post comment notification

### PR DESCRIPTION
**Pull request by joaquinto**
#### What does this PR do?

- Send a notification when a user post a comment
- Get all comments for a particular trip
- Delete a comment

#### Description of Task to be completed?

- Create Comment model
- Create comment migration
- Create comment seeder
- Create comment controller
- Create comment route
- Create comment middleware
- Create comment service

#### Any background context you want to provide?
-  Users or manager were not able to get notifications when a comment was made on the trip and they were not also able to delete their comments,  Hence, this PR enables staff to get notifications when a comment has been made on a trip and users can be able to delete their comments.

#### How should this be manually tested?

- clone the repo and cd into the root directory
- create a postgresql database
- `npm install`
- `npm run start:dev`
- use postman and test the following endpoints 
* POST  `/api/v1/trips/<:tripId>/comment`
* GET  `/api/v1/trips/<:tripId>/comments`
* DELETE  `/api/v1/trips/<:tripId>/comments/<:commentId>`

#### What are the relevant PT stories?
(#167727330)[https://www.pivotaltracker.com/story/show/167727330]

### CODE REVIEW

**Review by orley12 — COMMENTED** [review]
LGTM. well done bro

File: src/controllers/commentController.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  const data = {
+    tripId: Number(req.params.tripId),
+    userId: req.auth.id,
+    comment: req.body.comment
+  };
+  try {
+    const comment = await createComment(data);
+
+    // get the reason of the trip as title for the notification
+    const reason = await findTripById(Number(req.params.tripId));
```
Comment (Hector101) [inline src/controllers/commentController.js]: the variable naming `reason` lacks context, `tripReason` seems more appropriate.

File: src/controllers/commentController.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    // check if the comment was sent from the manager or the staff
+    const senderRole = (Number(reason.toJSON().userId) !== req.auth.id) ? 'Manager' : null;
```
Comment (Hector101) [inline src/controllers/commentController.js]: we can destructure `userId` and use it in multiple places where we have ` reason.toJSON().userId`

```
const { userId } = reason.toJSON();
```

Then `Number(userId)`

File: src/controllers/commentController.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  }
+};
+
+/**
+ * Delete a Comment
+ * @param {Object} req
+ * @param {Object} res
+ * @returns {Object} response object
+ */
+export const getAllComment = async (req, res) => {
```
Comment (Hector101) [inline src/controllers/commentController.js]: `getComments` seems more appropriate to `getAllComment`

File: src/controllers/commentController.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  try {
+    const comment = await getAllComments(Number(req.params.tripId));
```
Comment (Hector101) [inline src/controllers/commentController.js]: If `comment` is an array of comments then it should be renamed `comments`.

File: src/controllers/commentController.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    return !comment ? respondWithWarning(res, statusCode.resourceNotFound, 'Comments not found')
```
Comment (Hector101) [inline src/controllers/commentController.js]: This `!comment` is buggy. If no record is found, `comment` should be an empty array which is a truthy value, hence `respondWithWarning(res, statusCode.resourceNotFound, 'Comments not found')` will never be called.

Instead, check for `!comment.length`

**Review by Hector101 — APPROVED** [review]

**Pull request merged**